airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -86,25 +86,47 @@ def data_field(self) -> str:
 
 # Basic incremental stream
 class IncrementalShopifyStream(ShopifyStream, ABC):
-
     # Getting page size as 'limit' from parrent class
     @property
     def limit(self):
         return super().limit
 
+    # Setting the check point interval to the limit of the records output
     state_checkpoint_interval = limit
-
-    cursor_field = "id"
+    # Setting the default cursor field for all streams
+    cursor_field = "updated_at"
 
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        return {self.cursor_field: max(latest_record.get(self.cursor_field, 0), current_stream_state.get(self.cursor_field, 0))}
-
-    def request_params(self, stream_state=None, **kwargs):
-        stream_state = stream_state or {}
-        params = super().request_params(stream_state=stream_state, **kwargs)
-        params["since_id"] = stream_state.get(self.cursor_field)
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
+
+    def request_params(self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_state=stream_state, next_page_token=next_page_token, **kwargs)
+        if not next_page_token:
+            params["order"] = f"{self.order_field} asc"
+            if stream_state:
+                params[self.filter_field] = stream_state.get(self.cursor_field)
         return params
 
+    # Parse the stream_slice with respect to stream_state for Incremental refresh
+    def parse_stream_slice(self, stream_state: Mapping[str, Any] = None, records_slice: Mapping[str, Any] = None) -> Iterable[Mapping]:
```

Comment:
> So we can't send such API with such filter that it could return only fresh records. Is it possible to make such filter completely on API side?

The endpoint doesn't accept the updated_at filter, the only filers that accepted are limit, since_id, fields.

Reply:
Renamed to parse_records_slice.